libraries/ESP8266WebServer/src/detail/RequestHandlersImpl.h

```diff
@@ -125,7 +125,7 @@ class StaticRequestHandler : public RequestHandler<ServerType> {
         if (_cache_header.length() != 0)
             server.sendHeader("Cache-Control", _cache_header);
 
-        server.streamFile(f, contentType);
+        server.streamFile(f, contentType, requestMethod);
```

Comment:
Why not checking method before opening the file ?
(ie: move the File..open lines after cache header lines above,
 and surround file+streamfile block with `if method==HEAD`, no need to overload `::streamFile`)

Beside, in your version, `_streamFileCore` is called and sets content length which I'm not sure should be set in the `HEAD` case.

Reply:
Forget my above comment,
- content-length must be set to the same value with HEAD as compared to GET. 
- consequently file must be opened to know its size

ref:
```
The Content-Length entity-header field indicates the size of the entity-body,
in decimal number of OCTETs, sent to the recipient or, in the case of the
HEAD method, the size of the entity-body that would have been sent had
the request been a GET.
```

Reply:
As you can see I tried to make minimal changes only.
The HEAD is exactly the same as GET including headers and Content-Length. - there is just no body.
I can move the check on level up but the I suggest to rename _streamFileCore to _sendFileHeaders.